docs: add MDX component reference guide with visual examples (#3640)
neondatabase/website: content/docs/community/component-guide.md

---
title: Neon Documentation MDX Components Guide
subtitle: Most commonly used components for documentation writers
enableTableOfContents: true
isDraft: true
---

A practical guide for the most commonly used MDX components in Neon documentation. This guide focuses on components you'll use most frequently when writing documentation.

<InfoBlock>
<DocsList title="What you will learn:">
<p>How to use the most common MDX components in Neon documentation</p>

> <p>When to choose between different component types</p>

Reviewer: ```suggestion
<p>How to choose between different components</p>
```

<p>Best practices for component selection and usage</p>

> <p>Proper syntax and prop usage for each component</p>

Reviewer: ```suggestion
<p>Proper syntax and prop usage</p>
```

</DocsList>

<DocsList title="Related topics" theme="docs">
<a href="/docs/community/component-specialized">Component Specialized Guide</a>
<a href="/docs/community/component-icon-guide">Component Icon Guide</a>
<a href="/docs/community/component-architecture">Component Architecture</a>
<a href="/docs/community/contribution-guide">Documentation Contribution Guide</a>
</DocsList>
</InfoBlock>

## Quick Navigation

- [Essential Components](#essential-components) - Most commonly used
- [Tabbed Content](#tabbed-content) - CodeTabs and Tabs for organized content
- [Content Organization](#content-organization) - Structure and navigation components
- [Interactive Elements](#interactive-elements) - UI elements and forms
- [Common Shared Components](#common-shared-components) - Reusable content

---

> ## Essential Components

Reviewer: ```suggestion
## Essential components
```

These are the most frequently used components in Neon docs.

### Admonition

> Callouts for notes, warnings, and tips. 6 types available: `note` (default), `important`, `tip`, `info`, `warning`, `comingSoon`.

Reviewer: ```suggestion
Callouts for notes, warnings, and tips. There are six types available: `note` (default), `important`, `tip`, `info`, `warning`, `comingSoon`.
```

Use the API to create a database:

```bash
curl -X POST https://console.neon.tech/api/v2/projects/my-project/databases \
  -H "Authorization: Bearer $NEON_API_KEY"
```

</TabItem>
</Tabs>
````

**Live Preview:**

<Tabs labels={["Console", "CLI", "API"]}>
<TabItem>
Create a database using the Neon Console by navigating to your project dashboard and clicking "Create Database".
</TabItem>
<TabItem>
Use the Neon CLI to create a database:

```bash
neon databases create --name my-database
```

</TabItem>
<TabItem>
Use the API to create a database:

```bash
curl -X POST https://console.neon.tech/api/v2/projects/my-project/databases \
  -H "Authorization: Bearer $NEON_API_KEY"
```

</TabItem>
</Tabs>

---

## Content Organization

Components for structuring and organizing page content.

### TechCards / DetailIconCards

Technology cards with icons, titles, and descriptions. These components use different [icon systems](/docs/community/component-icon-guide) - see the [comparison table](#techcards-vs-detailiconcards-vs-docslist) below to choose the right one.

#### TechCards

Standard technology cards layout using [TechCards icons](/docs/community/component-icon-guide#techcards-icons):

> 

Reviewer: Any way we can fix the line breaks?

Author: It appears not as the Prettier job is reverting the changes!

```mdx
<TechCards>
  <a
    href="/docs/guides/node"
    title="Node.js"
    description="Connect Node.js applications to Neon"
    icon="node-js"
  >
    Node.js
  </a>
  <a
    href="/docs/guides/python"
    title="Python"
    description="Connect Python applications to Neon"
    icon="python"
  >
    Python
  </a>
  <a
    href="/docs/guides/nextjs"
    title="Next.js"
    description="Build Next.js apps with Neon"
    icon="next-js"
  >
    Next.js
  </a>
</TechCards>
```

**Live Preview:**

<TechCards>
<a href="/docs/guides/node" title="Node.js" description="Connect Node.js applications to Neon" icon="node-js">Node.js</a>
<a href="/docs/guides/python" title="Python" description="Connect Python applications to Neon" icon="python">Python</a>
<a href="/docs/guides/nextjs" title="Next.js" description="Build Next.js apps with Neon" icon="next-js">Next.js</a>
</TechCards>

---

#### DetailIconCards

Alternative layout using [DetailIconCards icons](/docs/community/component-icon-guide#detailiconcards-icons):

> 

Author: Fixed, thanks!

- Ideal for page introductions and overview sections

**Basic Two-Column Layout:**

```mdx
<InfoBlock>
<DocsList title="What you will learn:">
<p>How to view and modify data in the console</p>
<p>Create an isolated database copy per developer</p>
<p>Reset your branch to production when ready to start new work</p>
</DocsList>

<DocsList title="Related topics" theme="docs">
<a href="/docs/introduction/branching">About branching</a>
<a href="/docs/get-started-with-neon/workflow-primer">Branching workflows</a>
<a href="/docs/get-started-with-neon/connect-neon">Connect Neon to your stack</a>
</DocsList>
</InfoBlock>
```

**Renders as:**

<InfoBlock>
<DocsList title="What you will learn:">
<p>How to view and modify data in the console</p>
<p>Create an isolated database copy per developer</p>
<p>Reset your branch to production when ready to start new work</p>
</DocsList>

<DocsList title="Related topics" theme="docs">
<a href="/docs/introduction/branching">About branching</a>
<a href="/docs/get-started-with-neon/workflow-primer">Branching workflows</a>
<a href="/docs/get-started-with-neon/connect-neon">Connect Neon to your stack</a>
</DocsList>
</InfoBlock>

---

> ## Interactive Elements

Reviewer: ```suggestion
## Interactive elements
```

Components for user engagement and interaction.

### CheckList

Interactive checklists for setup guides and tutorials. CheckList uses CheckItem components internally.

```mdx

> <CheckList title="Setup Checklist">

Reviewer: ```suggestion
<CheckList title="Setup checklist">
```

  <CheckItem title="Create Neon account" href="#signup">
    Sign up for a free Neon account at console.neon.tech
  </CheckItem>
  <CheckItem title="Install dependencies" href="#install">
    Install the required packages for your project
  </CheckItem>
  <CheckItem title="Configure environment" href="#config">
    Set up your database connection string
  </CheckItem>
  <CheckItem title="Test connection" href="#test">
    Verify your application can connect to Neon
  </CheckItem>
</CheckList>
```

> **Live Preview:**

Reviewer: ```suggestion
**Live preview:**
```

</CheckList>

---

### CheckItem

Individual checklist items used within CheckList components.

```mdx
<CheckItem title="Task name" href="#anchor">
  Description of the task or requirement
</CheckItem>
```

---

**Usage Notes:**

- Always used within a `<CheckList>` component
- `title` prop is required
- `href` prop is optional for anchor linking
- Content is the description text

### CTA (Call to Action)

Prominent call-to-action buttons for important actions.

```mdx
<CTA
  title="Try Neon Free"
  description="Start building with serverless Postgres today. No credit card required."
  buttonText="Sign Up"
  buttonUrl="https://console.neon.tech/signup"
/>
```

**Live Preview:**

<CTA 
  title="Try Neon Free" 
  description="Start building with serverless Postgres today. No credit card required." 
  buttonText="Sign Up" 
  buttonUrl="https://console.neon.tech/signup" 
/>

---

### NeedHelp

Support widget for getting assistance.

```mdx
<NeedHelp />
```

**Live Preview:**

<NeedHelp />

---

---

## Common Shared Components

Reusable content components that load from shared templates.

### LinkAPIKey

Link to API key management in the console.

```mdx
<LinkAPIKey />
```

> **Live Preview:**

Reviewer: ```suggestion
**Live preview:**
```

<LinkAPIKey />

---

### FeatureBetaProps

Status indicator for beta features with custom feature name.

```mdx
<FeatureBetaProps feature_name="OpenTelemetry integration" />
```

> **Live Preview:**

Reviewer: ```suggestion
**Live preview:**
```
